### ISSUE TITLE
[BUG] Linter ignores span.End if called inside deferred function

### ISSUE THREAD

**Issue by bmon** [jjti/go-spancheck#12]
**Describe the bug**
I have some code that calls `span.End()` _inside_ a deferred function because I also want to trap and record panics via `recover()`.

Unfortunately this linter does not recognise that `span.End()` is called in the deferred function, and reports `span.End is not called on all paths, possible memory leak`

**To Reproduce**
```go
package main

import (
	"context"

	"go.opentelemetry.io/otel"
)

func _() {
	_, span := otel.Tracer("foo").Start(context.Background(), "bar")
	defer func() {
		span.End()
	}()
}
```

```
$ spancheck ./...
/home/brendan/git/go-spancheck/testdata/base/defer.go:10:2: span.End is not called on all paths, possible memory leak
/home/brendan/git/go-spancheck/testdata/base/defer.go:14:1: return can be reached without calling span.End
exit status 3
```

**Expected behavior**
The linter recognises this as a safe span completion that cannot be skipped and won't leak spans.
